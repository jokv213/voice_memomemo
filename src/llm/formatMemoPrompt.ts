interface MemoData {
  content: string;
  exercise?: string;
}

export function formatMemoPrompt(rawText: string, memoData: MemoData): string {
  return `あなたはフィットネスコーチのアシスタントです。音声から得られたトレーニングメモを読みやすく整理してください。

音声入力: "${rawText}"

関連種目: ${memoData.exercise || '不明'}
メモ内容: "${memoData.content}"

以下の観点で整理された文章を返してください:

1. 体調・感覚の記録
2. フォームや技術的な気づき
3. 今後の改善点や目標
4. その他の重要な観察

要求:
- 簡潔で読みやすい日本語
- 箇条書きまたは短い段落形式
- トレーニング日誌として有用な情報を強調
- 200文字以内で要約
- 不要な情報は省略`;
}
