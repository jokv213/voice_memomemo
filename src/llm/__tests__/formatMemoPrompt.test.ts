import {formatMemoPrompt} from '../formatMemoPrompt';

describe('formatMemoPrompt', () => {
  it('should generate prompt with exercise memo', () => {
    const rawText = 'ベンチプレスのフォームが良くなった気がする。次回はもう少し重量を上げてみよう';
    const memoData = {
      content: 'ベンチプレスのフォームが良くなった気がする。次回はもう少し重量を上げてみよう',
      exercise: 'ベンチプレス',
    };

    const result = formatMemoPrompt(rawText, memoData);

    expect(result).toMatchSnapshot();
    expect(result).toContain('ベンチプレスのフォームが良くなった気がする');
    expect(result).toContain('ベンチプレス');
    expect(result).toContain('体調・感覚');
    expect(result).toContain('フォームや技術');
    expect(result).toContain('改善点や目標');
  });

  it('should generate prompt with general training memo', () => {
    const rawText = '今日は調子が悪かった。肩が痛い';
    const memoData = {
      content: '今日は調子が悪かった。肩が痛い',
      exercise: undefined,
    };

    const result = formatMemoPrompt(rawText, memoData);

    expect(result).toMatchSnapshot();
    expect(result).toContain('今日は調子が悪かった。肩が痛い');
    expect(result).toContain('不明');
    expect(result).toContain('200文字以内');
  });

  it('should generate prompt with long memo content', () => {
    const rawText =
      'デッドリフトの練習をした。腰の位置に注意してフォームを確認。重量は前回より5kg増量できた。次回はさらに技術を磨いていきたい';
    const memoData = {
      content:
        'デッドリフトの練習をした。腰の位置に注意してフォームを確認。重量は前回より5kg増量できた。次回はさらに技術を磨いていきたい',
      exercise: 'デッドリフト',
    };

    const result = formatMemoPrompt(rawText, memoData);

    expect(result).toMatchSnapshot();
    expect(result).toContain('デッドリフト');
    expect(result).toContain('腰の位置');
    expect(result).toContain('簡潔で読みやすい');
  });
});
