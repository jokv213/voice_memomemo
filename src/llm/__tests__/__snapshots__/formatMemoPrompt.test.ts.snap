// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatMemoPrompt should generate prompt with exercise memo 1`] = `
"あなたはフィットネスコーチのアシスタントです。音声から得られたトレーニングメモを読みやすく整理してください。

音声入力: "ベンチプレスのフォームが良くなった気がする。次回はもう少し重量を上げてみよう"

関連種目: ベンチプレス
メモ内容: "ベンチプレスのフォームが良くなった気がする。次回はもう少し重量を上げてみよう"

以下の観点で整理された文章を返してください:

1. 体調・感覚の記録
2. フォームや技術的な気づき
3. 今後の改善点や目標
4. その他の重要な観察

要求:
- 簡潔で読みやすい日本語
- 箇条書きまたは短い段落形式
- トレーニング日誌として有用な情報を強調
- 200文字以内で要約
- 不要な情報は省略"
`;

exports[`formatMemoPrompt should generate prompt with general training memo 1`] = `
"あなたはフィットネスコーチのアシスタントです。音声から得られたトレーニングメモを読みやすく整理してください。

音声入力: "今日は調子が悪かった。肩が痛い"

関連種目: 不明
メモ内容: "今日は調子が悪かった。肩が痛い"

以下の観点で整理された文章を返してください:

1. 体調・感覚の記録
2. フォームや技術的な気づき
3. 今後の改善点や目標
4. その他の重要な観察

要求:
- 簡潔で読みやすい日本語
- 箇条書きまたは短い段落形式
- トレーニング日誌として有用な情報を強調
- 200文字以内で要約
- 不要な情報は省略"
`;

exports[`formatMemoPrompt should generate prompt with long memo content 1`] = `
"あなたはフィットネスコーチのアシスタントです。音声から得られたトレーニングメモを読みやすく整理してください。

音声入力: "デッドリフトの練習をした。腰の位置に注意してフォームを確認。重量は前回より5kg増量できた。次回はさらに技術を磨いていきたい"

関連種目: デッドリフト
メモ内容: "デッドリフトの練習をした。腰の位置に注意してフォームを確認。重量は前回より5kg増量できた。次回はさらに技術を磨いていきたい"

以下の観点で整理された文章を返してください:

1. 体調・感覚の記録
2. フォームや技術的な気づき
3. 今後の改善点や目標
4. その他の重要な観察

要求:
- 簡潔で読みやすい日本語
- 箇条書きまたは短い段落形式
- トレーニング日誌として有用な情報を強調
- 200文字以内で要約
- 不要な情報は省略"
`;
